app/models/v1/StreamReferenceData.scala

```diff
@@ -29,7 +31,7 @@ class StreamReferenceData(listName: ListName, metaData: MetaData) {
        |{
        |   "_links": {
        |     "self": {
-       |       "href": "${controllers.consumption.routes.ListRetrievalController.get(listName).url}"
+       |       "href": "${controllers.consumption.v1.routes.ListRetrievalController.get(listName).url.substring(5)}"
```

Comment:
@ajchilts what's the purpose of substring(5) here?

Reply:
@jackheslop96 - I guess its not so clear but its removing the version 'v1.0' or 'v2.0 from the front of the HATOES link as thats supplied in the header and does mot form part of the url.

Reply:
Ah I see. If you print out that url in the unit tests it doesn't actually appear to have the version in. I wonder if some kind of regex might be more airtight? Something like... (not sure if the full stop needs escaping or not)
```
val regex = """v(\d+).(\d+)/""".r
regex.replaceAllIn(controllers.consumption.v1.routes.ListRetrievalController.get(listName).url, "")
```
Perhaps the routes need reformatting a bit. They should also be prefixed with /test-only. Would `/test-only/customs-reference-data/v1.0/...` work?